include/core/Discretization.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+struct Discretization
+{
+    virtual ~Discretization() = default;
+
+    enum GridType { hydrofabric_locations, spatial_structured, spatial_unstructured };
```

Comment:
Do we expect these to end up one-to-one with concrete implementations of `Discretization`?